meta-mbl-distro/scripts/lib/mbl/update/versionedclassregistry.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (c) 2019 Arm Limited and Contributors. All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+"""
+Registry for keeping track of classes that represent versions of things.
+
+Each thing has a name and classes that represent different versions of that
+thing are identified by that name and an integer version.
+"""
+
+_registry = {}
+
+
+def register_versioned_class(name, version, cls):
+    """
+    Register a class as representing a version of a thing.
+
+    Args:
+    * name str: name of the thing that cls represents.
+    * version int: version of the thing that cls represents.
+    * cls class: class that represents the thing.
+
+    """
+    if name not in _registry:
+        _registry[name] = {}
+
+    assert version not in _registry[name]
+    _registry[name][version] = cls
```

Comment:
I'm wondering if we should use `weakref`s here so we don't prevent garbage collection of these objects?

Reply:
I don't think `weakref`s would help here because the things in the registry are required to create the objects that will have references to them.

We could potentially limit the scope of the registry so that it can be garbage collected once the `PayloadBuilder` has been created, but I don't think its worth it.